xmhf/src/xmhf-core/xmhf-runtime/xmhf-eventhub/arch/x86/vmx/peh-x86vmx-main.c

```diff
@@ -1133,11 +1133,8 @@ u32 xmhf_parteventhub_arch_x86vmx_intercept_handler(VCPU *vcpu, struct regs *r){
 				HALT_ON_ERRORCOND(0);
 			} else {
 				/* Inject NMI to guest */
-                vcpu->vmcs.control_VM_entry_exception_errorcode = 0;
-                vcpu->vmcs.control_VM_entry_interruption_information = NMI_VECTOR |
-                    INTR_TYPE_NMI |
-                    INTR_INFO_VALID_MASK;
-                printf("CPU(0x%02x): inject NMI\n", vcpu->id);
+                if(vcpu->vmx_guest_nmi_cfg.guest_nmi_enable)
```

Comment:
What will happen if this `if` is false? Looks like the NMI will be lost. If `if` can never be false, there should be a panic statement.